### PR TITLE
V1/33 account edit

### LINKED ISSUE #33: Account edit
Create vue.js component for user account edition

## User Story
As a __authenticated user__
I want to access an account edit page
So that I can change my information (eg: password, username, . . .)

- [x] define what must be on the component (refer to mockup), conditionnal component, forms,etc...
- [x] Create the component .Vue
- [x] Create the route to this component
- [x] Send data on the form to the backend
- [x] Test the components 

depends on: #59 
**Estimation: 2h**

### PR DESCRIPTION
**Pull request by SundayBraid**
related to #33 

Add Account editing form in Vue component,

- instead of using the real backend, we use MirageJs to handle request to a fake backend

### CODE REVIEW

File: src/public/css/app.css
Comment (kayoumido) [inline src/public/css/app.css]: This is a compiled file and it shouldn't be versioned by git. If you look in the `.gitignore`, you'll see that all the css & js files in the `public` folder, are ignored by git.

**Review by kayoumido — CHANGES_REQUESTED** [review]
Didn't really look at what you did just yet.

Once #105 is merged, rebase your branch and test your work with a fake api

File: src/resources/js/components/AccountEdition.vue
Comment (kayoumido) [inline src/resources/js/components/AccountEdition.vue]: Remvoe debug code

File: src/resources/js/components/AccountEdition.vue
Comment (kayoumido) [inline src/resources/js/components/AccountEdition.vue]: Same, remove

File: src/resources/js/App.vue
Comment (kayoumido) [inline src/resources/js/App.vue]: Add a better name. Something like `Edit profile` or you can find a French version since the app will be in Frenc

File: src/resources/js/App.vue
Comment (kayoumido) [inline src/resources/js/App.vue]: Is this normal that it's hardcoded?

File: src/public/css/app.css
Comment (SundayBraid) [reply src/public/css/app.css]: ok, i have removed it in a recent commit

File: src/resources/js/routes.js
Comment (kayoumido) [inline src/resources/js/routes.js]: The path should be something like `/user/edit`

File: src/resources/js/components/AccountEdition.vue
Comment (kayoumido) [inline src/resources/js/components/AccountEdition.vue:1]: This should be a single form instead of a form per field.

File: src/resources/js/components/AccountEdition.vue
Comment (kayoumido) [inline src/resources/js/components/AccountEdition.vue]: This is the method that needs to be called when submitting your single form.

File: src/resources/js/components/AccountEdition.vue
Comment (kayoumido) [inline src/resources/js/components/AccountEdition.vue:112]: You need to add a method to validate the format of the email.

File: src/resources/js/components/AccountEdition.vue
Comment (kayoumido) [inline src/resources/js/components/AccountEdition.vue]: With the single form approach, you can have a User object to contain the changes. `updatedUser`.

**Review by kayoumido — CHANGES_REQUESTED** [review]
`src/public/js/app.js` & `src/public/js/app.css` are compiled files and **shouldn't** be versioned by git. They're already in the `.gitignore`.

File: src/resources/js/App.vue
Comment (SundayBraid) [reply src/resources/js/App.vue]: yes, I can remove it since it no more a part of the jobs

File: src/resources/js/routes.js
Comment (SundayBraid) [reply src/resources/js/routes.js]: ok

File: src/resources/js/components/AccountEdition.vue
Comment (SundayBraid) [reply src/resources/js/components/AccountEdition.vue:1]: I find a form by data easier to handle with the associate function

File: src/resources/js/components/AccountEdition.vue
Comment (SundayBraid) [reply src/resources/js/components/AccountEdition.vue]: ok

File: src/resources/js/App.vue
Comment (SundayBraid) [reply src/resources/js/App.vue]: ok

File: src/resources/js/components/AccountEdition.vue
Comment (kayoumido) [inline src/resources/js/components/AccountEdition.vue]: TLD's can be greater than 3 characters. You should require a minimum of 2 and no max e.g. `(\.\w{2,})+`

**Review by kayoumido — CHANGES_REQUESTED** [review]
Remove `src/public/js/app.js` from Git

Apart for some linting issues (I'm working on setting it up) seems good.

File: src/resources/js/components/AccountEdition.vue
Comment (SundayBraid) [reply src/resources/js/components/AccountEdition.vue]: okay

File: src/resources/js/components/AccountEdition.vue
Comment (kayoumido) [inline src/resources/js/components/AccountEdition.vue]: The old value should be placed in a `value` attribute.

`The placeholder attribute specifies a short hint that describes the expected value of an input field` 

[Source](https://www.w3schools.com/tags/att_input_placeholder.asp)

File: src/resources/js/components/AccountEdition.vue
Comment (kayoumido) [inline src/resources/js/components/AccountEdition.vue]: same as UpdateFirstName

File: src/resources/js/components/AccountEdition.vue
Comment (kayoumido) [inline src/resources/js/components/AccountEdition.vue]: same as UpdateFirstName

File: src/resources/js/components/AccountEdition.vue
Comment (kayoumido) [inline src/resources/js/components/AccountEdition.vue]: same as UpdateFirstName

File: src/resources/js/components/AccountEdition.vue
Comment (kayoumido) [inline src/resources/js/components/AccountEdition.vue:10]: Add an `id` to the input it's bound to.

`the for attribute specifies which form element a label is bound to` 

[Source](https://www.w3schools.com/TAGS/att_for.asp)

**Review by kayoumido — CHANGES_REQUESTED** [review]

File: src/resources/js/components/AccountEdition.vue
Comment (SundayBraid) [reply src/resources/js/components/AccountEdition.vue]: I appears that the ``value`` attribut is already overridden by v-model, so i have remove the binding in placeholder.
the error output:
```
  :value="this.loggedUser.fname" conflicts with v-model on the same element because the latter already expands to a value binding internally
```

Comment (kayoumido) [reply src/resources/js/components/AccountEdition.vue]: Then in the data, you need to do something like this :
```js
      updateUserInfo: {
        fname: loggedUser.fname,
        lname:  loggedUser.lname,
        email: loggedUser.email,  
        password: ''
      }
```

File: src/resources/js/routes.js
Comment (fabio10000) [inline src/resources/js/routes.js]: this change belongs to another branch

File: src/resources/js/components/Hello.vue
Comment (fabio10000) [inline src/resources/js/components/Hello.vue]: not sure about the utility of this

File: src/resources/js/components/Authentication.vue
Comment (fabio10000) [inline src/resources/js/components/Authentication.vue]: this file belongs to another issue

File: src/resources/js/app.js
Comment (fabio10000) [inline src/resources/js/app.js]: This file shouldn't appear on your pr

File: src/resources/js/App.vue
Comment (fabio10000) [inline src/resources/js/App.vue]: this file shouldn't appear on your pr

File: src/package.json
Comment (fabio10000) [inline src/package.json:18]: this file shouldn't be in your PR

File: src/app/Http/Resources/UserResource.php
Comment (fabio10000) [inline src/app/Http/Resources/UserResource.php:1]: This file shouldn't be in your PR

File: src/app/Http/Resources/QuestionResource.php
Comment (fabio10000) [inline src/app/Http/Resources/QuestionResource.php]: This file shouldn't be in your PR

File: src/app/Http/Resources/PollResource.php
Comment (fabio10000) [inline src/app/Http/Resources/PollResource.php]: This file shouldn't be in your PR

File: src/app/Http/Resources/ConferenceResource.php
Comment (fabio10000) [inline src/app/Http/Resources/ConferenceResource.php]: This file shouldn't be in your PR

File: src/app/Http/Resources/ChoiceResource.php
Comment (fabio10000) [inline src/app/Http/Resources/ChoiceResource.php]: This file shouldn't be in your PR

File: src/app/Http/Controllers/API/PresentationController.php
Comment (fabio10000) [inline src/app/Http/Controllers/API/PresentationController.php]: This file shouldn't be in your PR

**Review by fabio10000 — CHANGES_REQUESTED** [review]
Do not include useless files on your PR this should result in painfull merges and extreme headaches.
If you need files from dev you should do a rebase to keep the git history instead of copy past the code or whatever you've done.

File: src/resources/js/routes.js
Comment (SundayBraid) [reply src/resources/js/routes.js]: right I removed it in [d672321](https://github.com/kayoumido/HEIGVD-PRO-A-11/pull/103/commits/d67232131843572a5c41096a8fc56328204dc54d)

File: src/resources/js/components/Hello.vue
Comment (SundayBraid) [reply src/resources/js/components/Hello.vue]: right,I removed it in [4c5cb31](https://github.com/kayoumido/HEIGVD-PRO-A-11/pull/103/commits/4c5cb31c4ac943b0bde044ec922eb037e45bec3c)

File: src/resources/js/components/Authentication.vue
Comment (SundayBraid) [reply src/resources/js/components/Authentication.vue]: right,I removed it in [4c5cb31](https://github.com/kayoumido/HEIGVD-PRO-A-11/pull/103/commits/4c5cb31c4ac943b0bde044ec922eb037e45bec3c)

File: src/resources/js/app.js
Comment (SundayBraid) [reply src/resources/js/app.js]: right,I removed it in [4c5cb31](https://github.com/kayoumido/HEIGVD-PRO-A-11/pull/103/commits/4c5cb31c4ac943b0bde044ec922eb037e45bec3c)

File: src/resources/js/App.vue
Comment (SundayBraid) [reply src/resources/js/App.vue]: right,I removed it in [4c5cb31](https://github.com/kayoumido/HEIGVD-PRO-A-11/pull/103/commits/4c5cb31c4ac943b0bde044ec922eb037e45bec3c)

File: src/app/Http/Resources/QuestionResource.php
Comment (SundayBraid) [reply src/app/Http/Resources/QuestionResource.php]: right,I removed it in [4c5cb31](https://github.com/kayoumido/HEIGVD-PRO-A-11/pull/103/commits/4c5cb31c4ac943b0bde044ec922eb037e45bec3c)

File: src/app/Http/Resources/PollResource.php
Comment (SundayBraid) [reply src/app/Http/Resources/PollResource.php]: right,I removed it in [4c5cb31](https://github.com/kayoumido/HEIGVD-PRO-A-11/pull/103/commits/4c5cb31c4ac943b0bde044ec922eb037e45bec3c)

File: src/app/Http/Resources/ConferenceResource.php
Comment (SundayBraid) [reply src/app/Http/Resources/ConferenceResource.php]: right,I removed it in [4c5cb31](https://github.com/kayoumido/HEIGVD-PRO-A-11/pull/103/commits/4c5cb31c4ac943b0bde044ec922eb037e45bec3c)

File: src/app/Http/Resources/ChoiceResource.php
Comment (SundayBraid) [reply src/app/Http/Resources/ChoiceResource.php]: right,I removed it in [4c5cb31](https://github.com/kayoumido/HEIGVD-PRO-A-11/pull/103/commits/4c5cb31c4ac943b0bde044ec922eb037e45bec3c)

File: src/app/Http/Controllers/API/PresentationController.php
Comment (SundayBraid) [reply src/app/Http/Controllers/API/PresentationController.php]: right,I removed it in [4c5cb31](https://github.com/kayoumido/HEIGVD-PRO-A-11/pull/103/commits/4c5cb31c4ac943b0bde044ec922eb037e45bec3c)

File: src/resources/js/components/AccountEdition.vue
Comment (alfavre) [inline src/resources/js/components/AccountEdition.vue]: you forgot the ; at the end of this line (159)

File: src/resources/js/routes.js
Comment (alfavre) [inline src/resources/js/routes.js:14]: the path should be in french, as our app is in french, and that's something the user can see.

**Review by alfavre — COMMENTED** [review]
Except some nitpicks, this seems okay for me.

**Review by fabio10000 — APPROVED** [review]

File: src/resources/js/routes.js
Comment (kayoumido) [reply src/resources/js/routes.js:14]: no

File: src/resources/js/components/AccountEdition.vue
Comment (SundayBraid) [reply src/resources/js/components/AccountEdition.vue]: thank, corrected on [d130657](https://github.com/kayoumido/HEIGVD-PRO-A-11/pull/103/commits/d130657dcd577b335ce38fdc84b47409ba52fa15)

**Review by kayoumido — APPROVED** [review]
All Gucci! As we discussed, we wont merge this branch. So I'll let you move the code when you'll start implementing the styleguide

**Comment by kayoumido** [comment]
Will be merged with #109

**Pull request closed without merging**
